diff --git a/src/drivers/net/efi/snpnet.c b/src/drivers/net/efi/snpnet.c
index 69ec6f5e554e7b39f0fab73d0b800ec2034dcb13..bcd4610e6bbff0e83f938d6af5c33c7ef28cf58f 100644
--- a/src/drivers/net/efi/snpnet.c
+++ b/src/drivers/net/efi/snpnet.c
@@ -311,12 +311,6 @@ static int snpnet_rx_filters ( struct net_device *netdev ) {
 	struct snp_nic *snp = netdev->priv;
 	UINT32 filters[] = {
 		snp->snp->Mode->ReceiveFilterMask,
-		( EFI_SIMPLE_NETWORK_RECEIVE_UNICAST |
-		  EFI_SIMPLE_NETWORK_RECEIVE_MULTICAST |
-		  EFI_SIMPLE_NETWORK_RECEIVE_BROADCAST ),
-		( EFI_SIMPLE_NETWORK_RECEIVE_UNICAST |
-		  EFI_SIMPLE_NETWORK_RECEIVE_PROMISCUOUS_MULTICAST |
-		  EFI_SIMPLE_NETWORK_RECEIVE_BROADCAST ),
 		( EFI_SIMPLE_NETWORK_RECEIVE_UNICAST |
 		  EFI_SIMPLE_NETWORK_RECEIVE_BROADCAST ),
 		( EFI_SIMPLE_NETWORK_RECEIVE_UNICAST ),
@@ -328,7 +322,7 @@ static int snpnet_rx_filters ( struct net_device *netdev ) {
 	/* Try possible receive filters in turn */
 	for ( i = 0; i < ( sizeof ( filters ) / sizeof ( filters[0] ) ); i++ ) {
 		efirc = snp->snp->ReceiveFilters ( snp->snp, filters[i],
-				EFI_SIMPLE_NETWORK_RECEIVE_MULTICAST, TRUE,
+				0, FALSE,
 				0, NULL );
 		if ( efirc == 0 )
 			return 0;
diff --git a/src/interface/efi/efi_snp.c b/src/interface/efi/efi_snp.c
index 088a3fdbda485ed705a1cd84100f08f56928dd6b..34e17a3c296ea5b1f100db263f1e24342f79db68 100644
--- a/src/interface/efi/efi_snp.c
+++ b/src/interface/efi/efi_snp.c
@@ -127,7 +127,6 @@ static void efi_snp_set_mode ( struct efi_snp_device *snpdev ) {
 	mode->MediaHeaderSize = ll_protocol->ll_header_len;
 	mode->MaxPacketSize = netdev->mtu;
 	mode->ReceiveFilterMask = ( EFI_SIMPLE_NETWORK_RECEIVE_UNICAST |
-				    EFI_SIMPLE_NETWORK_RECEIVE_MULTICAST |
 				    EFI_SIMPLE_NETWORK_RECEIVE_BROADCAST );
 	assert ( ll_addr_len <= sizeof ( mode->CurrentAddress ) );
 	memcpy ( &mode->CurrentAddress, netdev->ll_addr, ll_addr_len );
